### ISSUE TITLE
Arch Linux Crash on open, when open no books...

### ISSUE THREAD

**Issue by Flavor-name** [crosswire/xiphos#1161]
Opening Xiphos with Terminal I get this error...
Program is unusable.
`$ xiphos

(process:10842): Gtk-WARNING **: 11:23:21.304: Locale not supported by C library.
	Using the fallback 'C' locale.
Fontconfig warning: ignoring LC_CTYPE=C;LC_NUMERIC=en_IL;LC_TIME=en_IL;LC_COLLATE=C;LC_MONETARY=en_IL;LC_MESSAGES=C;LC_PAPER=en_IL;LC_NAME=en_IL;LC_ADDRESS=en_IL;LC_TELEPHONE=en_IL;LC_MEASUREMENT=en_IL;LC_IDENTIFICATION=en_IL: not a valid region tag
Segmentation fault (core `dumped)`

**Comment by alerque** [comment]
It looks like you have misconfigured your system's locale. What do you have as the output of `localectl`?

**Comment by Flavor-name** [comment]
This is the results you asked of me:

'
$ localectl
System Locale: LANG=en_IL
    VC Keymap: us
   X11 Layout: us
'
'
$ grep -vE '^#' /etc/locale.gen
en_US.UTF-8 UTF-8  

en_IL UTF-8
'

The problem still occurs.
The problem seemed to occur after Kerenl 6.9.5

I have tried uninstalling, rebooting, reinstalling. Didn't fix the issue.
Not downloading bibles, but copying the hidden .sword folder to the home directory after fresh install, made the program function, opening instead of crashing, but still no books were loaded.
In the AUR, there was a recent build, May of 2024. 
I noticed the error condition in late June, could this be related?
I have always had my locale set this way, there has never been a problem previously.
Need any other system commands to sauce it out, i'm checking back more often.

**Comment by alerque** [comment]
I don't think your locales are formatted right.

First I would edit your `/etc/locale.gen` file to replace the IL related line to this:

```
en_IL.UTF-8 UTF-8
```

Then regenerate locales:

```console
$ sudo locale-gen
```

Next use that as your system locale:

```console
$ sudo localectl set-locale en_IL.UTF-8
```

That should stick after a reboot, but I'd double check anyway with `localectl` again after a reboot and login.

If that doesn't work, can you additionally check that Xiphos does cooperate with an en_US locale:

```console
$ env LANG=en_US.UTF-8 xiphos
```

**Comment by Flavor-name** [comment]
My ```/etc/locale.gen``` contains this entry:

```#en_IL UTF-8```

There is no period.
There is no other choice.

Is the program expecting to find a period seperated entry?
There are many entries in my .gen file that do not use a ```.UTF-8```.

Will modifying the entry by adding a period affect the system adversly?
I'll check archlinux.org to see if the locale.gen file they maintain is different in anyway.

**Comment by alerque** [comment]
Not only is the entry not a correct match for the current locale database, it is also commented out (the leading `#`) on the line. You should put the entry in like I suggested and then regenerate the local system locale cache as I noted in my previous message.

Your locale.gen file has obviously been edited from the original if it only has that one commented out line. Also that may have been the right format at one point, but it is not any more. The entries must match what the database records them as for it to work. The `locale.gen` file does not get overwritten on system updates if you have ever modified it my hand, hence how it may have old entries. In this case you probably have a `locale.gen.packnew` file sitting around that you never dealt with.

**Comment by alerque** [comment]
By the way if you don't use any other locales you can safely nuke *everything* in `locale.gen` and *just* have the one line with this entry:

```
en_IL.UTF-8 UTF-8
```

The vast majority of stuff in that file is just commented out possible entries people might want to use. You do not need anything at all in the file except the locales you really want to use. For example all my systems have this:

```console
$ cat /etc/locale.gen
en_US.UTF-8 UTF-8
ru_RU.UTF-8 UTF-8
tr_TR.UTF-8 UTF-8
```

That's it, and those are locales I actually want any programs on my system to be localized in. Ones that don't support those locales will fall back to something they do support anyway.

**Comment by Flavor-name** [comment]
I have a fresh install, which only has about one month of use. I don't alter these files.
I choose my location in the gui setup screen, keyboard and locale are the first choices in the install process.
The files I have are those generated by the install process.
The only maintenance I do is to add a hebrew keyboard for my Bible research.
I update using `pacman -Syu`
I don't go tweaking `/etc/xyz` files.
Is it possible the Distro provider has created a non standard locale set???

**Comment by Flavor-name** [comment]
```
$ cat /etc/locale.gen
# Configuration file for locale-gen
#
# lists of locales that are to be generated by the locale-gen command.
#
# Each line is of the form:
#
#     <locale> <charset>
#
#  where <locale> is one of the locales given in /usr/share/i18n/locales
#  and <charset> is one of the character sets listed in /usr/share/i18n/charmaps
#
#  The locale-gen command will generate all the locales,
#  placing them in /usr/lib/locale.
#
#  A list of supported locales is given in /usr/share/i18n/SUPPORTED
#  and is included in this file. Uncomment the needed locales below.
#
#aa_DJ.UTF-8 UTF-8  
#aa_DJ ISO-8859-1  
#aa_ER UTF-8  
#aa_ER@saaho UTF-8  
#aa_ET UTF-8  
#af_ZA.UTF-8 UTF-8  
#af_ZA ISO-8859-1  
#agr_PE UTF-8  
#ak_GH UTF-8  
#am_ET UTF-8  
#an_ES.UTF-8 UTF-8  
#an_ES ISO-8859-15  
#anp_IN UTF-8  
#ar_AE.UTF-8 UTF-8  
#ar_AE ISO-8859-6  
#ar_BH.UTF-8 UTF-8  
#ar_BH ISO-8859-6  
#ar_DZ.UTF-8 UTF-8  
#ar_DZ ISO-8859-6  
#ar_EG.UTF-8 UTF-8  
#ar_EG ISO-8859-6  
#ar_IN UTF-8  
#ar_IQ.UTF-8 UTF-8  
#ar_IQ ISO-8859-6  
#ar_JO.UTF-8 UTF-8  
#ar_JO ISO-8859-6  
#ar_KW.UTF-8 UTF-8  
#ar_KW ISO-8859-6  
#ar_LB.UTF-8 UTF-8  
#ar_LB ISO-8859-6  
#ar_LY.UTF-8 UTF-8  
#ar_LY ISO-8859-6  
#ar_MA.UTF-8 UTF-8  
#ar_MA ISO-8859-6  
#ar_OM.UTF-8 UTF-8  
#ar_OM ISO-8859-6  
#ar_QA.UTF-8 UTF-8  
#ar_QA ISO-8859-6  
#ar_SA.UTF-8 UTF-8  
#ar_SA ISO-8859-6  
#ar_SD.UTF-8 UTF-8  
#ar_SD ISO-8859-6  
#ar_SS UTF-8  
#ar_SY.UTF-8 UTF-8  
#ar_SY ISO-8859-6  
#ar_TN.UTF-8 UTF-8  
#ar_TN ISO-8859-6  
#ar_YE.UTF-8 UTF-8  
#ar_YE ISO-8859-6  
#ayc_PE UTF-8  
#az_AZ UTF-8  
#az_IR UTF-8  
#as_IN UTF-8  
#ast_ES.UTF-8 UTF-8  
#ast_ES ISO-8859-15  
#be_BY.UTF-8 UTF-8  
#be_BY CP1251  
#be_BY@latin UTF-8  
#bem_ZM UTF-8  
#ber_DZ UTF-8  
#ber_MA UTF-8  
#bg_BG.UTF-8 UTF-8  
#bg_BG CP1251  
#bhb_IN.UTF-8 UTF-8  
#bho_IN UTF-8  
#bho_NP UTF-8  
#bi_VU UTF-8  
#bn_BD UTF-8  
#bn_IN UTF-8  
#bo_CN UTF-8  
#bo_IN UTF-8  
#br_FR.UTF-8 UTF-8  
#br_FR ISO-8859-1  
#br_FR@euro ISO-8859-15  
#brx_IN UTF-8  
#bs_BA.UTF-8 UTF-8  
#bs_BA ISO-8859-2  
#byn_ER UTF-8  
#ca_AD.UTF-8 UTF-8  
#ca_AD ISO-8859-15  
#ca_ES.UTF-8 UTF-8  
#ca_ES ISO-8859-1  
#ca_ES@euro ISO-8859-15  
#ca_ES@valencia UTF-8  
#ca_FR.UTF-8 UTF-8  
#ca_FR ISO-8859-15  
#ca_IT.UTF-8 UTF-8  
#ca_IT ISO-8859-15  
#ce_RU UTF-8  
#chr_US UTF-8  
#ckb_IQ UTF-8  
#cmn_TW UTF-8  
#crh_UA UTF-8  
#cs_CZ.UTF-8 UTF-8  
#cs_CZ ISO-8859-2  
#csb_PL UTF-8  
#cv_RU UTF-8  
#cy_GB.UTF-8 UTF-8  
#cy_GB ISO-8859-14  
#da_DK.UTF-8 UTF-8  
#da_DK ISO-8859-1  
#de_AT.UTF-8 UTF-8  
#de_AT ISO-8859-1  
#de_AT@euro ISO-8859-15  
#de_BE.UTF-8 UTF-8  
#de_BE ISO-8859-1  
#de_BE@euro ISO-8859-15  
#de_CH.UTF-8 UTF-8  
#de_CH ISO-8859-1  
#de_DE.UTF-8 UTF-8  
#de_DE ISO-8859-1  
#de_DE@euro ISO-8859-15  
#de_IT.UTF-8 UTF-8  
#de_IT ISO-8859-1  
#de_LI.UTF-8 UTF-8  
#de_LU.UTF-8 UTF-8  
#de_LU ISO-8859-1  
#de_LU@euro ISO-8859-15  
#doi_IN UTF-8  
#dsb_DE UTF-8  
#dv_MV UTF-8  
#dz_BT UTF-8  
#el_GR.UTF-8 UTF-8  
#el_GR ISO-8859-7  
#el_GR@euro ISO-8859-7  
#el_CY.UTF-8 UTF-8  
#el_CY ISO-8859-7  
#en_AG UTF-8  
#en_AU.UTF-8 UTF-8  
#en_AU ISO-8859-1  
#en_BW.UTF-8 UTF-8  
#en_BW ISO-8859-1  
#en_CA.UTF-8 UTF-8  
#en_CA ISO-8859-1  
#en_DK.UTF-8 UTF-8  
#en_DK ISO-8859-1  
#en_GB.UTF-8 UTF-8  
#en_GB ISO-8859-1  
#en_HK.UTF-8 UTF-8  
#en_HK ISO-8859-1  
#en_IE.UTF-8 UTF-8  
#en_IE ISO-8859-1  
#en_IE@euro ISO-8859-15  
#en_IL UTF-8  
#en_IN UTF-8  
#en_NG UTF-8  
#en_NZ.UTF-8 UTF-8  
#en_NZ ISO-8859-1  
#en_PH.UTF-8 UTF-8  
#en_PH ISO-8859-1  
#en_SC.UTF-8 UTF-8  
#en_SG.UTF-8 UTF-8  
#en_SG ISO-8859-1  
en_US.UTF-8 UTF-8  
#en_US ISO-8859-1  
#en_ZA.UTF-8 UTF-8  
#en_ZA ISO-8859-1  
#en_ZM UTF-8  
#en_ZW.UTF-8 UTF-8  
#en_ZW ISO-8859-1  
#eo UTF-8  
#es_AR.UTF-8 UTF-8  
#es_AR ISO-8859-1  
#es_BO.UTF-8 UTF-8  
#es_BO ISO-8859-1  
#es_CL.UTF-8 UTF-8  
#es_CL ISO-8859-1  
#es_CO.UTF-8 UTF-8  
#es_CO ISO-8859-1  
#es_CR.UTF-8 UTF-8  
#es_CR ISO-8859-1  
#es_CU UTF-8  
#es_DO.UTF-8 UTF-8  
#es_DO ISO-8859-1  
#es_EC.UTF-8 UTF-8  
#es_EC ISO-8859-1  
#es_ES.UTF-8 UTF-8  
#es_ES ISO-8859-1  
#es_ES@euro ISO-8859-15  
#es_GT.UTF-8 UTF-8  
#es_GT ISO-8859-1  
#es_HN.UTF-8 UTF-8  
#es_HN ISO-8859-1  
#es_MX.UTF-8 UTF-8  
#es_MX ISO-8859-1  
#es_NI.UTF-8 UTF-8  
#es_NI ISO-8859-1  
#es_PA.UTF-8 UTF-8  
#es_PA ISO-8859-1  
#es_PE.UTF-8 UTF-8  
#es_PE ISO-8859-1  
#es_PR.UTF-8 UTF-8  
#es_PR ISO-8859-1  
#es_PY.UTF-8 UTF-8  
#es_PY ISO-8859-1  
#es_SV.UTF-8 UTF-8  
#es_SV ISO-8859-1  
#es_US.UTF-8 UTF-8  
#es_US ISO-8859-1  
#es_UY.UTF-8 UTF-8  
#es_UY ISO-8859-1  
#es_VE.UTF-8 UTF-8  
#es_VE ISO-8859-1  
#et_EE.UTF-8 UTF-8  
#et_EE ISO-8859-1  
#et_EE.ISO-8859-15 ISO-8859-15  
#eu_ES.UTF-8 UTF-8  
#eu_ES ISO-8859-1  
#eu_ES@euro ISO-8859-15  
#fa_IR UTF-8  
#ff_SN UTF-8  
#fi_FI.UTF-8 UTF-8  
#fi_FI ISO-8859-1  
#fi_FI@euro ISO-8859-15  
#fil_PH UTF-8  
#fo_FO.UTF-8 UTF-8  
#fo_FO ISO-8859-1  
#fr_BE.UTF-8 UTF-8  
#fr_BE ISO-8859-1  
#fr_BE@euro ISO-8859-15  
#fr_CA.UTF-8 UTF-8  
#fr_CA ISO-8859-1  
#fr_CH.UTF-8 UTF-8  
#fr_CH ISO-8859-1  
#fr_FR.UTF-8 UTF-8  
#fr_FR ISO-8859-1  
#fr_FR@euro ISO-8859-15  
#fr_LU.UTF-8 UTF-8  
#fr_LU ISO-8859-1  
#fr_LU@euro ISO-8859-15  
#fur_IT UTF-8  
#fy_NL UTF-8  
#fy_DE UTF-8  
#ga_IE.UTF-8 UTF-8  
#ga_IE ISO-8859-1  
#ga_IE@euro ISO-8859-15  
#gd_GB.UTF-8 UTF-8  
#gd_GB ISO-8859-15  
#gez_ER UTF-8  
#gez_ER@abegede UTF-8  
#gez_ET UTF-8  
#gez_ET@abegede UTF-8  
#gl_ES.UTF-8 UTF-8  
#gl_ES ISO-8859-1  
#gl_ES@euro ISO-8859-15  
#gu_IN UTF-8  
#gv_GB.UTF-8 UTF-8  
#gv_GB ISO-8859-1  
#ha_NG UTF-8  
#hak_TW UTF-8  
#he_IL.UTF-8 UTF-8  
#he_IL ISO-8859-8  
#hi_IN UTF-8  
#hif_FJ UTF-8  
#hne_IN UTF-8  
#hr_HR.UTF-8 UTF-8  
#hr_HR ISO-8859-2  
#hsb_DE ISO-8859-2  
#hsb_DE.UTF-8 UTF-8  
#ht_HT UTF-8  
#hu_HU.UTF-8 UTF-8  
#hu_HU ISO-8859-2  
#hy_AM UTF-8  
#hy_AM.ARMSCII-8 ARMSCII-8  
#ia_FR UTF-8  
#id_ID.UTF-8 UTF-8  
#id_ID ISO-8859-1  
#ig_NG UTF-8  
#ik_CA UTF-8  
#is_IS.UTF-8 UTF-8  
#is_IS ISO-8859-1  
#it_CH.UTF-8 UTF-8  
#it_CH ISO-8859-1  
#it_IT.UTF-8 UTF-8  
#it_IT ISO-8859-1  
#it_IT@euro ISO-8859-15  
#iu_CA UTF-8  
#ja_JP.EUC-JP EUC-JP  
#ja_JP.UTF-8 UTF-8  
#ka_GE.UTF-8 UTF-8  
#ka_GE GEORGIAN-PS  
#kab_DZ UTF-8  
#kk_KZ.UTF-8 UTF-8  
#kk_KZ PT154  
#kl_GL.UTF-8 UTF-8  
#kl_GL ISO-8859-1  
#km_KH UTF-8  
#kn_IN UTF-8  
#ko_KR.EUC-KR EUC-KR  
#ko_KR.UTF-8 UTF-8  
#kok_IN UTF-8  
#ks_IN UTF-8  
#ks_IN@devanagari UTF-8  
#ku_TR.UTF-8 UTF-8  
#ku_TR ISO-8859-9  
#kw_GB.UTF-8 UTF-8  
#kw_GB ISO-8859-1  
#ky_KG UTF-8  
#lb_LU UTF-8  
#lg_UG.UTF-8 UTF-8  
#lg_UG ISO-8859-10  
#li_BE UTF-8  
#li_NL UTF-8  
#lij_IT UTF-8  
#ln_CD UTF-8  
#lo_LA UTF-8  
#lt_LT.UTF-8 UTF-8  
#lt_LT ISO-8859-13  
#lv_LV.UTF-8 UTF-8  
#lv_LV ISO-8859-13  
#lzh_TW UTF-8  
#mag_IN UTF-8  
#mai_IN UTF-8  
#mai_NP UTF-8  
#mfe_MU UTF-8  
#mg_MG.UTF-8 UTF-8  
#mg_MG ISO-8859-15  
#mhr_RU UTF-8  
#mi_NZ.UTF-8 UTF-8  
#mi_NZ ISO-8859-13  
#miq_NI UTF-8  
#mjw_IN UTF-8  
#mk_MK.UTF-8 UTF-8  
#mk_MK ISO-8859-5  
#ml_IN UTF-8  
#mn_MN UTF-8  
#mni_IN UTF-8  
#mnw_MM UTF-8  
#mr_IN UTF-8  
#ms_MY.UTF-8 UTF-8  
#ms_MY ISO-8859-1  
#mt_MT.UTF-8 UTF-8  
#mt_MT ISO-8859-3  
#my_MM UTF-8  
#nan_TW UTF-8  
#nan_TW@latin UTF-8  
#nb_NO.UTF-8 UTF-8  
#nb_NO ISO-8859-1  
#nds_DE UTF-8  
#nds_NL UTF-8  
#ne_NP UTF-8  
#nhn_MX UTF-8  
#niu_NU UTF-8  
#niu_NZ UTF-8  
#nl_AW UTF-8  
#nl_BE.UTF-8 UTF-8  
#nl_BE ISO-8859-1  
#nl_BE@euro ISO-8859-15  
#nl_NL.UTF-8 UTF-8  
#nl_NL ISO-8859-1  
#nl_NL@euro ISO-8859-15  
#nn_NO.UTF-8 UTF-8  
#nn_NO ISO-8859-1  
#nr_ZA UTF-8  
#nso_ZA UTF-8  
#oc_FR.UTF-8 UTF-8  
#oc_FR ISO-8859-1  
#om_ET UTF-8  
#om_KE.UTF-8 UTF-8  
#om_KE ISO-8859-1  
#or_IN UTF-8  
#os_RU UTF-8  
#pa_IN UTF-8  
#pa_PK UTF-8  
#pap_AW UTF-8  
#pap_CW UTF-8  
#pl_PL.UTF-8 UTF-8  
#pl_PL ISO-8859-2  
#ps_AF UTF-8  
#pt_BR.UTF-8 UTF-8  
#pt_BR ISO-8859-1  
#pt_PT.UTF-8 UTF-8  
#pt_PT ISO-8859-1  
#pt_PT@euro ISO-8859-15  
#quz_PE UTF-8  
#raj_IN UTF-8  
#rif_MA UTF-8  
#ro_RO.UTF-8 UTF-8  
#ro_RO ISO-8859-2  
#ru_RU.KOI8-R KOI8-R  
#ru_RU.UTF-8 UTF-8  
#ru_RU ISO-8859-5  
#ru_UA.UTF-8 UTF-8  
#ru_UA KOI8-U  
#rw_RW UTF-8  
#sa_IN UTF-8  
#sah_RU UTF-8  
#sat_IN UTF-8  
#sc_IT UTF-8  
#sd_IN UTF-8  
#sd_IN@devanagari UTF-8  
#se_NO UTF-8  
#sgs_LT UTF-8  
#shn_MM UTF-8  
#shs_CA UTF-8  
#si_LK UTF-8  
#sid_ET UTF-8  
#sk_SK.UTF-8 UTF-8  
#sk_SK ISO-8859-2  
#sl_SI.UTF-8 UTF-8  
#sl_SI ISO-8859-2  
#sm_WS UTF-8  
#so_DJ.UTF-8 UTF-8  
#so_DJ ISO-8859-1  
#so_ET UTF-8  
#so_KE.UTF-8 UTF-8  
#so_KE ISO-8859-1  
#so_SO.UTF-8 UTF-8  
#so_SO ISO-8859-1  
#sq_AL.UTF-8 UTF-8  
#sq_AL ISO-8859-1  
#sq_MK UTF-8  
#sr_ME UTF-8  
#sr_RS UTF-8  
#sr_RS@latin UTF-8  
#ss_ZA UTF-8  
#st_ZA.UTF-8 UTF-8  
#st_ZA ISO-8859-1  
#sv_FI.UTF-8 UTF-8  
#sv_FI ISO-8859-1  
#sv_FI@euro ISO-8859-15  
#sv_SE.UTF-8 UTF-8  
#sv_SE ISO-8859-1  
#sw_KE UTF-8  
#sw_TZ UTF-8  
#syr UTF-8  
#szl_PL UTF-8  
#ta_IN UTF-8  
#ta_LK UTF-8  
#tcy_IN.UTF-8 UTF-8  
#te_IN UTF-8  
#tg_TJ.UTF-8 UTF-8  
#tg_TJ KOI8-T  
#th_TH.UTF-8 UTF-8  
#th_TH TIS-620  
#the_NP UTF-8  
#ti_ER UTF-8  
#ti_ET UTF-8  
#tig_ER UTF-8  
#tk_TM UTF-8  
#tl_PH.UTF-8 UTF-8  
#tl_PH ISO-8859-1  
#tn_ZA UTF-8  
#to_TO UTF-8  
#tpi_PG UTF-8  
#tr_CY.UTF-8 UTF-8  
#tr_CY ISO-8859-9  
#tr_TR.UTF-8 UTF-8  
#tr_TR ISO-8859-9  
#ts_ZA UTF-8  
#tt_RU UTF-8  
#tt_RU@iqtelif UTF-8  
#ug_CN UTF-8  
#uk_UA.UTF-8 UTF-8  
#uk_UA KOI8-U  
#unm_US UTF-8  
#ur_IN UTF-8  
#ur_PK UTF-8  
#uz_UZ.UTF-8 UTF-8  
#uz_UZ ISO-8859-1  
#uz_UZ@cyrillic UTF-8  
#ve_ZA UTF-8  
#vi_VN UTF-8  
#wa_BE ISO-8859-1  
#wa_BE@euro ISO-8859-15  
#wa_BE.UTF-8 UTF-8  
#wae_CH UTF-8  
#wal_ET UTF-8  
#wo_SN UTF-8  
#xh_ZA.UTF-8 UTF-8  
#xh_ZA ISO-8859-1  
#yi_US.UTF-8 UTF-8  
#yi_US CP1255  
#yo_NG UTF-8  
#yue_HK UTF-8  
#yuw_PG UTF-8  
#zh_CN.GB18030 GB18030  
#zh_CN.GBK GBK  
#zh_CN.UTF-8 UTF-8  
#zh_CN GB2312  
#zh_HK.UTF-8 UTF-8  
#zh_HK BIG5-HKSCS  
#zh_SG.UTF-8 UTF-8  
#zh_SG.GBK GBK  
#zh_SG GB2312  
#zh_TW.EUC-TW EUC-TW  
#zh_TW.UTF-8 UTF-8  
#zh_TW BIG5  
#zu_ZA.UTF-8 UTF-8  
#zu_ZA ISO-8859-1  

###
#
# Locales enabled by Calamares
en_IL UTF-8
```

**Comment by Flavor-name** [comment]
Looks like my installer is causing the locale issue.
::sigh::

**Comment by alerque** [comment]
Okay now we're getting somewhere!

So first of all there was some miscommunication. You said "contains this entry […] There is no other choice", by which I understood that was the only line in the file. In retrospect I assume you mean "the only line related to en_IL", but that is where I assumed you must have edited the file because by default it contains a large list of mostly commented possible entries.

Second, there is clearly a bug in Calamares. This is the first time I've even heard of that installer. There is certainly nothing wrong with using a 3rd party installer especially since Arch Linux doesn't even officially *have* a graphical installer. But it does mean this is is not an Arch Linux bug, it's a big in the installer. I would definitely raise this as an issue on their tracker (wherever that is). The installer should be placing well formed locales according to the current glibc version's locale database.

Third, that still leaves something in the way of Xiphos bugs or at least undesirable behavior.

1. Xiphos shouldn't crash completely, even if/when the system locale is malformed.
1. There is no reason at all to even warn in the event of a regional fallback. We don't need a translation for `en_IL`, it should just fall back to `en_US` or `en_GB` or whatever it has for missing regional variants. It should only throw an obnoxious warning for "no matching locale" if the base language is different.
1. The warning message shown in the screen shot above doesn't even surface the relevant "en_IL" info as it should now that the locale is actually set correctly.

I'd say we should keep this issue open to track fixing those 3 related things in Xiphos, but again definitely do raise an issue to track the problem with Calamares separately.

**Comment by alerque** [comment]
P.S. It looks like Calmares has [an issue tracker here](https://github.com/calamares/calamares/issues).

**Comment by nwalkewicz** [comment]
> I have followed your recommended steps, except for US_UTF-8 only. I wanted to see if the resolution would be en_IL.UTF-8. I followed the `locale.gen` recommendation, then rebooted. Next I enable `localectl` and rebooted. Hineh ah-knee.
> 
> Something is fixed, but not completely. Including a screen shot of the current issue. ![xiphos-error-screenshot](https://private-user-images.githubusercontent.com/155650245/350769820-6082f1ef-6389-469e-8786-0d2f28dd34db.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NTU0MTMsIm5iZiI6MTcyMTk1NTExMywicGF0aCI6Ii8xNTU2NTAyNDUvMzUwNzY5ODIwLTYwODJmMWVmLTYzODktNDY5ZS04Nzg2LTBkMmYyOGRkMzRkYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNlQwMDUxNTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMzRlOGU1MzZkOWY2ZDc1MzAxZTc1NzI0Mzc0Zjg3NDI0ZTBkNTdmZTcyZjc0MjM1NTJmYjE0N2Q0NjI2YWU5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.55C3ZvSDxwRMzzazS7dpo5UKtv6dv14_TJQRoOX43tc)

For a workaround, go to: Edit > Preferences > Options (under "General") > Special Locale. First, look through the list and manually set it to your locale. Click Close and restart Xiphos. That error shouldn't appear again. (I had the same exact problem for a while, I thought it was something weird I did while building from source.)

I just now set mine back to "None" instead of "en_US" and it seems to be working fine. Assuming this is how it works, it should now follow my system locale if it's ever changed.

**Comment by alerque** [comment]
Add one more action item to my list:

4. The "None" option should be renamed to make it clear that this is what to pick to match the system.

**Comment by nwalkewicz** [comment]
I like that idea. Maybe rename "Special Locale" to just "Locale" and change "None" to "System Default"? Might be worth its own issue.

**Comment by Flavor-name** [comment]
There were other issues with the application, like, Strong's Concordance wouldn't function when clicking a number.

There must have been a bad update from one of the dependencies, glibc or such. I'm not sure when exactly, but it was involving multiple applications not just Xiphos.

I have solved the issue by reinstalling the OS.
While this is not the best solution  to discover the root of the problem, it was the fastest solution. With fresh installs on a new system, the applications works without issue.

There is one exception though, WLC module and OSHB Module, though correctly loaded when looking in the .sword folders, show squares instead of Fonts. 
The Module Installer stated these modules "Failed Install" but yet were present in the application.
I have downloaded copies of the modules and manually copied the files to their respective directories.
I am back to being able to use the application, yeah.
